### PR TITLE
server/cache: wrap some methods in clusterInfo

### PR DESCRIPTION
**Pull request by huachaohuang**
This commit is separated from https://github.com/pingcap/pd/pull/359, it
remove some unnecessary code and wrap some code in clusterInfo.

### CODE REVIEW

**Comment by huachaohuang** [comment]
PTAL @siddontang @overvenus

File: server/cache.go [hunk 1]
```diff
@@ -22,11 +22,16 @@ import (
...
 	"github.com/juju/errors"
 	"github.com/ngaut/log"
-	raftpb "github.com/pingcap/kvproto/pkg/eraftpb"
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/kvproto/pkg/pdpb"
 )
 
+var (
+	errStoreNotFound = func(storeID uint64) error {
+		return errors.Errorf("store not found: %v", storeID)
```
Comment (siddontang) [inline server/cache.go]: store %d not found

File: server/cache.go [hunk 2]
```diff
@@ -277,53 +282,21 @@ func (r *regionsInfo) heartbeatVersion(region *metapb.Region) (bool, *metapb.Reg
...
 	cacheRegion := r.regions[region.GetId()]
 	if err := checkStaleRegion(cacheRegion, region); err != nil {
-		return nil, false, errors.Trace(err)
+		return false, errors.Trace(err)
 	}
 
 	if region.GetRegionEpoch().GetConfVer() > cacheRegion.GetRegionEpoch().GetConfVer() {
 		// ConfChanged, update
 		r.updateRegion(region)
-		return r.getChangePeer(cacheRegion, region), true, nil
```
Comment (siddontang) [inline server/cache.go]: where to handle ChangePeer now ?

Comment (huachaohuang) [inline server/cache.go]: It's not necessary to handle it. The original code just records the `ChangePeer` event, but that's already recorded when the corresponding operator has finished, there is no reason to record this event here again.

Comment (siddontang) [inline server/cache.go]: Got it

**Comment by siddontang** [comment]
Rest LGTM

**Pull request merged**